addons/pos_self_order/static/src/pages/order_cart/order_cart.js

```diff
@@ -24,9 +25,15 @@ export class OrderCart extends Component {
         return this.selfOrder.self_order_mode === "each" ? "Pay" : "Order";
     }
 
-    processOrder() {
+    async processOrder() {
+        if (this.sendInProgress) {
+            return;
+        }
+
         if (this.selfOrder.self_order_mode === "meal") {
-            this.selfOrder.sendDraftOrderToServer();
+            this.sendInProgress = true;
+            await this.selfOrder.sendDraftOrderToServer();
+            this.sendInProgress = false;
```

Comment:
```suggestion
            try {
                await this.selfOrder.sendDraftOrderToServer();
            } finally {
                this.sendInProgress = false;
            }
```

Need to make sure it's set to `false` also when `sendDraftOrderToServer` throws an error.

Reply:
good catch ! thx